compiler/test/runnable/string_literal_merge.d

```diff
@@ -0,0 +1,14 @@
+/**
+COMPILE_SEPARATELY:
+EXTRA_SOURCES: imports/string_literal_merge_imp.d
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=24286
+
+extern(C) string getHello();
+
+extern(C) int main()
+{
+    assert(getHello.ptr == "hello".ptr);
```

Comment:
This test is likely to be at best dubious as we're not relying fully on the compiler to do make this work.